GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -187,6 +208,39 @@ - (void)removeEvents:(NSSet<NSNumber *> *)eventIDs {
   });
 }
 
+- (void)libraryDataForKey:(nonnull NSString *)key
+               onComplete:(nonnull void (^)(NSData *_Nullable))onComplete {
+  dispatch_async(_storageQueue, ^{
+    NSString *dataPath = [[[self class] libraryDataPath] stringByAppendingPathComponent:key];
+    NSData *data = [NSData dataWithContentsOfFile:dataPath];
+    if (onComplete) {
+      onComplete(data);
+    }
+  });
+}
+
+- (void)storeLibraryData:(nullable NSData *)data
+                  forKey:(nonnull NSString *)key
+              onComplete:(nonnull void (^)(NSError *_Nullable error))onComplete {
+  dispatch_async(_storageQueue, ^{
+    NSError *error;
+    NSString *dataPath = [[[self class] libraryDataPath] stringByAppendingPathComponent:key];
+    if (data && data.length > 0) {
+      [data writeToFile:dataPath options:NSDataWritingAtomic error:&error];
+      if (onComplete) {
+        onComplete(error);
+      }
+    } else {
+      if ([[NSFileManager defaultManager] fileExistsAtPath:dataPath]) {
```

Comment:
Would the API be easier to use if there was a `removeLibraryData` method instead of a zero lengthed `NSData`?

Either way, please add a test for this path.

Reply:
Done.